File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Nodejs/core/handler/GetStaticFieldHandler.js
const AbstractHandler = require("./AbstractHandler");

class GetStaticFieldHandler extends AbstractHandler {
  requiredParametersCount = 2;

  constructor() {
    super();
  }

  process(command) {
    try {
      if (command.payload.length < this.requiredParametersCount) {
        throw new Error("Array Static Field parameters mismatch");
      }
      const { payload } = command;
      let type = payload[0];
      let field = payload[1];

      let staticField = type[field];
      if (typeof staticField === "undefined") {
        let fields = Object.keys(type);
        let message = `Field ${field} not found in class. Available fields:\n`;
        fields.forEach((fieldIter) => {
          message += `${fieldIter}\n`;
        });
        throw new Error(message);
      } else {
        return staticField;
      }
    } catch (error) {
      throw this.process_stack_trace(error, this.constructor.name);
    }
  }
}

module.exports = new GetStaticFieldHandler();

File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Ruby/Windows/X64/core/handler/get_static_field_handler.rb
require_relative 'abstract_command_handler'

class GetStaticFieldHandler < AbstractCommandHandler
  def initialize
    @required_parameters_count = 2
  end

  def process(command)
    get_static_field(command)
  end

  def get_static_field(command)
    begin
      if command.payload.length != @required_parameters_count
        raise ArgumentError.new "Static field parameters mismatch"
      end

      merged_value = '@@' + command.payload[1]
      if command.payload[0].class_variable_defined?(merged_value)
        response = command.payload[0].class_variable_get(merged_value)
      else
        response = command.payload[0].const_get(command.payload[1])
      end
      return response
    rescue NameError
      fields = command.payload[0].class_variables
      message = "Field #{command.payload[1]} not found in class #{command.payload[0].name}. Available fields:\n"
      fields.each { |field_iter| message += "#{field_iter}\n" }
      raise Exception, message
    rescue Exception => e
      return e
    end
  end
end
